### ISSUE TITLE
Improve S3Select Performance in alerts api for large numbers of files close in time

### ISSUE THREAD

**Issue by rleighton** [panther-labs/panther#2397]
### Description

If there are 100 of thousands of files in a short timespan for an alert the alerts api times out due to the listing time (we list serially).

NOTE: we may want to wait on this until appsync is removed, since we can have more than a 30sec timeout and this is a corner case, waiting a min or two could be ok.

Suggested approach:

- raise alerts lambda size from 512 to 2048 to get more speed and concurrency
- raise s3 select concurrency to 100
- refactor s3 select search to list concurrently
 - this can be done by driving the s3select worker go routines via a channel using concurrent s3 listings, where each listing divides the time span of the alert (or starting by next token) by the concurrency and searches each range segment at the same time. Since we want the first N returned, this change means we have to wait until all listings are done, sort results (cannot stop early).

### Acceptance Criteria

- Alerts API does not timeout on a test of 200k objects searched

**Comment by kostaspap** [comment]
It seems that there is a bug in the rules-engine and that bug has been there since a while. We split the logs to separate files depending on dedup - we shouldn't be doing that. It's a one-line fix and will likely drastically improve the alerts-api performance.

**Comment by kostaspap** [comment]
Closing in favor of https://github.com/panther-labs/panther/issues/2536
